File: homework3/src/main/java/com/chocomiruku/homework3/task1/Book.java
package com.chocomiruku.homework3.task1;

import static com.chocomiruku.homework3.Util.formatPrice;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class Book implements Publication {

    private final double price;
    private final int wordCount;

    public Book(double price, int wordCount) {
        this.price = price;
        this.wordCount = wordCount;
    }

    @Override
    public String getType() {
        if (wordCount <= 0) throw new IllegalArgumentException("WordCount should be positive");

        if (wordCount > 7500) {
            return "Novel";
        } else if (wordCount > 1000) {
            return "Short Story";
        }
        return "Flash Fiction";
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (!(obj instanceof Book)) {
            return false;
        }

        Book other = (Book) obj;
        return other.price == this.price && other.wordCount == this.wordCount;
    }

    @NonNull
    @Override
    public String toString() {
        return "Type: " + getType() + "\n" + "Price: " + formatPrice(price) + ", Words: " + wordCount;
    }

    @Override
    public double getPrice() {
        return this.price;
    }
}

File: app/src/main/java/com/chocomiruku/homework1/task1/Book.kt
package com.chocomiruku.homework1.task1

import com.chocomiruku.homework1.formatPrice

class Book(override val price: Double, override val wordCount: Int) : Publication {
    override fun getType(): String {
        return when {
            wordCount in 1..1000 -> "Flash Fiction"
            wordCount in 1001..7500 -> "Short Story"
            wordCount <= 0 -> throw IllegalArgumentException("WordCount should be positive")
            else -> "Novel"
        }
    }

    override fun equals(other: Any?): Boolean {
        return other is Book && other.price == this.price && other.wordCount == this.wordCount
    }

    override fun toString(): String {
        return "Type: " + getType() + "\n" + "Price: " + formatPrice(price) + ", Words: " + wordCount
    }
}